src/store/actionCreators/saveSets.ts

```diff
@@ -0,0 +1,62 @@
+import { ThunkAction } from 'redux-thunk';
+import {
+  FAILURE_CREATE,
+  RE_INITIALIZE_STATE,
+  SaveSetParams,
+  SaveSetsActionTypes,
+  TAG_NAME_CONFLICT,
+  TOGGLE_PENDING_CREATE,
+} from '../saveSetTypes';
+import { saveSetCountForTag } from '../../services/sets';
+// @ts-ignore
+import saveSet from '@kfarranger/components/dist/utils/saveSet';
+import { RootState } from '../rootState';
+import graphql from '../../services/arranger';
+
+export const createSaveSet = (
+  payload: SaveSetParams,
+): ThunkAction<void, RootState, null, SaveSetsActionTypes> => async (dispatch) => {
+  const { tag, userId, api, sqon, onSuccess, onNameConflict } = payload;
+
+  dispatch(togglePendingCreate(true));
+  try {
+    const tagIsUnique = (await saveSetCountForTag(tag, userId)) === 0;
+    if (tagIsUnique) {
+      await saveSet({
+        type: 'participant',
+        path: 'kf_id',
+        sqon,
+        userId: userId,
+        api: graphql(api),
+        tag: tag,
+      });
+      onSuccess();
```

Comment:
I would check if the callback is available. If that is to be reused elsewhere nothing says we would need to use a callback - it's optional